### PR TITLE
LLC public API polishing

### PR DESCRIPTION
**Pull request by evsaev**
**This PR**:
- updates `UserListQuery` to follow the naming convention (+ filter scope)
- updates `ChannelListQuery` to follow the naming convention (+ filter scope)
- updates `ChannelMemberListQuery` to follow the naming convention (+ filter scope)
- updates `ChannelQuery` to follow the naming convention
- fixes visability scope for `JSONEncoder/JSONDecoder/Data` extensions

### CODE REVIEW

File: Sources_v3/StreamChat/Models/Attachment.swift [hunk 1]
```diff
@@ -38,14 +38,23 @@ extension _ChatMessageAttachment {
...
+        ///   - type: The attachment type. Attachment rendering will be chosen based on it type.
+        ///   - extraData: The extra data which can be used to include additional information about the attachment
+        /// used in client application.
         public init(
             localURL: URL,
-            fileName: String,
+            fileName: String? = nil,
             type: AttachmentType,
-            extraData: ExtraData.Attachment
+            extraData: ExtraData.Attachment = .defaultValue
```
Comment (VojtaStavik) [inline Sources_v3/StreamChat/Models/Attachment.swift:54]: 👍

File: Sources_v3/StreamChat/Utils/Codable+Extensions.swift [hunk 2]
```diff
@@ -44,19 +44,19 @@ extension JSONDecoder {
 
 extension JSONEncoder {
     /// A default `JSONEncoder`.
-    public static var `default`: JSONEncoder = stream
+    static var `default`: JSONEncoder = stream
```
Comment (VojtaStavik) [inline Sources_v3/StreamChat/Utils/Codable+Extensions.swift:47]: 👏

File: Sources_v3/StreamChat/Database/DataStore.swift [hunk 3]
```diff
@@ -31,6 +31,8 @@ public struct DataStore<ExtraData: ExtraDataTypes> {
     /// - Returns: If there's a user object in the locally cached data matching the provided `id`, returns the matching
     /// model object. If a user object doesn't exist locally, returns `nil`.
     ///
+    /// **Warning**: Can be called on the `main` thread only.
+    ///
```
Comment (VojtaStavik) [inline Sources_v3/StreamChat/Database/DataStore.swift:35]: _should_ be called

**Review by VojtaStavik — APPROVED** [review]
Great changes Pavel, thank you!

File: Sources_v3/StreamChat/Query/ChannelQuery.swift [hunk 4]
```diff
@@ -1,11 +1,20 @@
 //
-// Copyright © 2020 Stream.io Inc. All rights reserved.
+// Copyright © 2021 Stream.io Inc. All rights reserved.
 //
 
 import Foundation
 
+public typealias ChannelQuery = _ChannelQuery<NoExtraData>
```
[the remark this answers is not in the document]
Comment (evsaev) [reply Sources_v3/StreamChat/Query/ChannelQuery.swift:15]: fixed in 56db2e16a46a9e5d576149158270491a23960306

File: Sources_v3/StreamChat/Database/DataStore.swift (hunk 3 above)
Comment (evsaev) [reply Sources_v3/StreamChat/Database/DataStore.swift]: fixed in 0b20b8ae9b15ed9f6bda42e5fa2919c5e187a5b6

**Pull request merged**